### ISSUE TITLE
sequelize.sync does not create id column

### ISSUE THREAD

**Issue by Shtirlits** [sequelize/sequelize#5587]
TL;DR `sequelize.sync({force: true})` does not create all columns, while `Model.sync({force: true})` does

I have models and mocha tests for them.
Here is `sequelize`
```coffee
Sequelize = require 'sequelize'
sequelize = new Sequelize 'db_name', `username`, `password`,
  host: 'localhost'
  dialect: 'postgres'
  port: 5432
  logging: false
  define:
    timestamps: false
```

Here is `Model`:
```coffee
module.exports = (sequelize, DataTypes) ->
  sequelize.define 'Model',
    id:
      type: DataTypes.BIGINT
      primaryKey: true
      autoIncrement: true
      validate:
        isInt:
          msg: 'Id must be integer'

    personId:
      type: DataTypes.INTEGER
      field: 'person_id'
      allowNull: false
      validate:
        isInt: msg: 'Person id must be integer'

    personRoleId:
      type: DataTypes.INTEGER
      field: 'person_role_id'
      allowNull: false
      validate:
        isInt: msg: 'Person role id must be integer'
  ,
    tableName: 'table_name'
```

Here is test (sync part and the part, which fails):
```coffee
describe 'Model', ->
  it 'was synchronized', (done) ->
    Model
      .sync force: true
      .then ->
        done()
      , (err) ->
        done err

  it 'has all fields', ->
    Model
      .findAll
        logging: (sql) ->
          assert.equal 'Executing (default): SELECT "id", "person_id" AS "personId", "person_role_id" AS "personRoleId" FROM "table_name" AS "Model";', sql
```

In the test section I have such error
```
SequelizeDatabaseError: column "id" does not exist
    at [object Object].Query.formatError (node_modules/sequelize/lib/dialects/postgres/query.js:357:14)
    at [object Object].<anonymous> (node_modules/sequelize/lib/dialects/postgres/query.js:88:19)
    at [object Object].Query.handleError (node_modules/pg/lib/query.js:108:8)
    at [object Object].<anonymous> (node_modules/pg/lib/client.js:171:26)
    at Socket.<anonymous> (node_modules/pg/lib/connection.js:109:12)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)
```

If i have `sequelize.sync({force: true})` in the test - `id` column is not created for some reason (other two columns are), and the test fails (no `sync` errors are thrown, just the error I've pasted above). If I have `Model.sync({force: true})` - all fields are created, but if I have an empty database - other tests are failing because of associations.

**Comment by mickhansen** [comment]
`sequelize.sync` calls `Model.sync` so that's highly odd.
Try logging the createa table statements by sync

**Comment by Shtirlits** [comment]
sql for `sequelize.sync` (related to this model):

```sql
Executing (default): DROP TABLE IF EXISTS "table_name" CASCADE;

Executing (default): DROP TABLE IF EXISTS "table_name" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "table_name" ("person_role_id" INTEGER  REFERENCES "person_roles" ("id") ON DELETE CASCADE ON UPDATE CASCADE, "person_id" INTEGER  REFERENCES "people" ("id") ON DELETE CASCADE ON UPDATE CASCADE, PRIMARY KEY ("person_role_id","person_id"));
Executing (default): SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'table_name' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;
```

sql for `Model.sync`
```sql
Executing (default): DROP TABLE IF EXISTS "table_name" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "table_name" ("id"  SERIAL , "person_id" INTEGER NOT NULL, "person_role_id" INTEGER NOT NULL, PRIMARY KEY ("id"));
Executing (default): SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'table_name' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;
```

As you can see - no mention of `id` in the first case

**Comment by mickhansen** [comment]
Is the model used as the `through` argument of a relation?
Or do you have a `through: 'tableString'` that clashes with the same model name as that model?

**Comment by Shtirlits** [comment]
Yes I do. I another model I have:

```coffee
  @belongsToMany models.Role,
    through: 'people_person_roles'
    foreignKey: 'person_id'
    as: 'personRoles'
```

`people_person_roles` is the real table name for `Model`

**Comment by mickhansen** [comment]
Right, that's your issue. You're creating two models when doing that. If you're defining a model for your through table, then you should also use that model as the through argument.

**Comment by Shtirlits** [comment]
Unfortunately, that did not help, error message is the same. This is what I have now:
```coffee
@belongsToMany models.Role,
  through: Model
  foreignKey: 'person_id'
  as: 'personRoles'
```

corresponding sql:
```sql
Executing (default): DROP TABLE IF EXISTS "people_person_roles" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "people_person_roles" ("person_role_id" INTEGER  REFERENCES "person_roles" ("id") ON DELETE CASCADE ON UPDATE CASCADE, "PersonId" INTEGER  REFERENCES "people" ("id") ON DELETE CASCADE ON UPDATE CASCADE, PRIMARY KEY ("person_role_id","PersonId"));
Executing (default): SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'people_person_roles' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;
```

Also, not sure from where that pascal cased `PersonId` came instead of `person_id` after passing `Model` as a parameter

**Comment by mickhansen** [comment]
It would be helpfull if you could post all 3 of your actual models and the relations between them.
The way you have it defined now it should create with a id primary key.

PersonId is the auto generated one, should not be there when using the foreignKey/otherKey properly.

**Comment by Shtirlits** [comment]
Yeah, missed one unfixed `through`. Now it works, thanks!

**Comment by mickhansen** [comment]
Great :)

**Comment by gurudanito001** [comment]
The solution is to add defaultValue to the id declaration,

{
      type: DataTypes.UUID,
      primaryKey: true,
      unique: true,
      allowNull: false,
      defaultValue: DataTypes.UUIDV4 // Or DataTypes.UUIDV1
}
